tests/unit/NetworkTest.js

```diff
@@ -32,6 +32,7 @@ beforeEach(() => {
     MainQueue.clear();
     HttpUtils.cancelPendingRequests();
     PersistedRequests.clear();
+    NetworkStore.checkRequiredData();
```

Comment:
Why is this change here?

Reply:
Oh I thought thats part of it cc @tienifr

Reply:
In https://github.com/Expensify/App/blob/88644166f51c9f7ea8211df0b9afee1ca4784700/src/libs/Network/NetworkStore.js#L28 we're using resolveIsReadyPromise to run resolve when this page is loaded and we call `resolveIsReadyPromise` to execute resolve function in `checkRequiredData`. 

 https://github.com/Expensify/App/blob/88644166f51c9f7ea8211df0b9afee1ca4784700/tests/unit/NetworkTest.js#L44 
we create a new promise and a new resolve function => we have to call `checkRequiredData` again in beforeEach.

- Why does the previous version works?
before each test run, we call `Onyx.clear()` that update subscriber even the data doesn't change => checkRequiredData is called
https://github.com/Expensify/App/blob/88644166f51c9f7ea8211df0b9afee1ca4784700/src/libs/Network/NetworkStore.js#L57

Reply:
@mountiny @roryabraham What do you think about my explanation above?

Reply:
@mountiny @roryabraham Any updates?

Reply:
Thanks for explaining